src/gluonts/model/evaluation.py

```diff
@@ -0,0 +1,180 @@
+    input_: DataEntry,
+    label: DataEntry,
+    forecast: Forecast,
+    seasonality: Optional[int] = None,
+    mask_invalid_label: bool = True,
+    allow_nan_forecast: bool = False,
+) -> ChainMap:
+    forecast_dict = BatchForecast(forecast, allow_nan=allow_nan_forecast)
+
+    freq = forecast.start_date.freqstr
+    if seasonality is None:
+        seasonality = get_seasonality(freq=freq)
+
+    label_target = label["target"]
+    if mask_invalid_label:
+        label_target = np.ma.masked_invalid(label_target)
+
+    other_data = {
+        "label": np.expand_dims(label_target, axis=0),
+        "seasonal_error": np.expand_dims(
+            seasonal_error(
+                input_["target"], seasonality=seasonality, time_axis=-1
+            ),
+            axis=0,
+        ),
+    }
+
+    return ChainMap(other_data, forecast_dict)
+
+
+def evaluate(
+    metrics,
+    test_data: TestData,
+    forecasts: Iterable[Forecast],
+    *,
+    axis: Optional[Union[int, tuple]] = None,
+    mask_invalid_label: bool = True,
+    allow_nan_forecast: bool = False,
+    seasonality: Optional[int] = None
+) -> dict:
+    """
+    Evaluate ``metrics`` by comparing ``forecasts`` to ``test_data``.
+
+    The optional ``axis`` arguments controls aggregation of the metrics:
+    - ``None`` (default) aggregates across all dimensions
+    - ``0`` aggregates across the batch dimension
+    - ``1`` aggregates across the time dimension
+    - ``2`` aggregates across the target dimension
+
+    Return results as a dictionary.
+    """
+    label_shape = first(test_data.label)["target"].shape
+
+    assert len(label_shape) in [1, 2]
+
+    if axis is None:
+        axis = tuple(range(len(label_shape) + 1))
+    if isinstance(axis, int):
+        axis = (axis,)
+
+    assert axis is None or all(a in range(3) for a in axis)
+
+    evaluators = {}
+    for metric in metrics:
+        evaluator = metric(axis=axis)
+        evaluators[evaluator.name] = evaluator
+
+    index_data = []
+
+    for input_, label, forecast in zip(
+        test_data.input, test_data.label, forecasts
+    ):
+        if 0 not in axis:
+            index_data.append((forecast.item_id, forecast.start_date))
+
+        data_batch = _get_data_batch(
+            input_,
+            label,
+            forecast,
+            seasonality=seasonality,
+            mask_invalid_label=mask_invalid_label,
+            allow_nan_forecast=allow_nan_forecast,
+        )
+
+        for evaluator in evaluators.values():
+            evaluator.update(data_batch)
+
+    metrics_dict = {
+        metric_name: evaluator.get()
+        for metric_name, evaluator in evaluators.items()
+    }
+
+    if index_data:
+        metrics_dict["__index_0"] = index_data
+
+    return metrics_dict
+
+
+def evaluate_pd(*args, **kwargs) -> pd.DataFrame:
+    """
+    Call ``evaluate``, flatten the results and wrap them
+    in a ``pd.DataFrame``.
+    """
+    metrics_dict = evaluate(*args, **kwargs)
```

Comment:
I'm really not a fan of Hungarian notation.

If you want to indicate that this is a dict, I think this is better:

```suggestion
    metrics: dict = evaluate(*args, **kwargs)
```

Reply:
done